core/node/libp2p/routingopt.go

```diff
@@ -23,6 +24,64 @@ type RoutingOption func(
 	...peer.AddrInfo,
 ) (routing.Routing, error)
 
+// Default HTTP routers used in parallel to DHT when Routing.Type = "auto"
+var defaultHTTPRouters = []string{
+	"https://dev.cid.contact/routing/v1", // TODO: remove when cid.contact is ready
+	"https://cid.contact/routing/v1",     // https://github.com/ipfs/kubo/issues/9422#issuecomment-1338142084
+	// TODO: add an independent router from Cloudflare
+}
+
+// ConstructDefaultRouting returns routers used when Routing.Type is unset or set to "auto"
+func ConstructDefaultRouting(peerID string, addrs []string, privKey string) func(
+	ctx context.Context,
+	host host.Host,
+	dstore datastore.Batching,
+	validator record.Validator,
+	bootstrapPeers ...peer.AddrInfo,
+) (routing.Routing, error) {
+	return func(
+		ctx context.Context,
+		host host.Host,
+		dstore datastore.Batching,
+		validator record.Validator,
+		bootstrapPeers ...peer.AddrInfo,
+	) (routing.Routing, error) {
+		// Defined routers will be queried in parallel (optimizing for response speed)
+		// Different trade-offs can be made by setting Routing.Type = "custom" with own Routing.Routers
+		var routers []*routinghelpers.ParallelRouter
+
+		// Run the default DHT routing (same as Routing.Type = "dht")
+		dhtRouting, err := DHTOption(ctx, host, dstore, validator, bootstrapPeers...)
+		if err != nil {
+			return nil, err
+		}
+		routers = append(routers, &routinghelpers.ParallelRouter{
+			Router:       dhtRouting,
+			IgnoreError:  false,           // TODO: confirm this is what we want
+			Timeout:      1 * time.Minute, // TODO: what value makes sense here?
+			ExecuteAfter: 0,
+		})
```

Comment:
:question: 
I don't know what to do with `Timeout` – `Routing.Type=dht` afaik does not have such thing, or it is somewhere inside of go-lib2p code.

@ajnavarro  thoughts?

Reply:
3s may be not enough for slower networks, causing false-negatives. Set to 15s for now, we adjust later.